src/lib/defaultExtractor.js

```diff
@@ -19,7 +19,7 @@ export function defaultExtractor(content) {
 function* buildRegExps() {
   yield regex.pattern([
     // Variants
-    /((?=([^\s"'\\\[]+:))\2)?/,
+    /((?=([^\s"'\\]+:))\2)?/,
```

Comment:
This will require some work to make it exactly right

Reply:
lol yep I was thinking about this. I'll throw this change against my new candidate refactor to see what breaks…

I had already tested support for arbitrary variant syntax (at least detection wise) in the form of [foo]:bar

Reply:
ngl I love this idea so much

Reply:
I pushed a regex tweak to fix the tests 👍👍